javascript/javascript1/week2/student_manager.js

```diff
@@ -0,0 +1,34 @@
+const class07Students = [];
+function addStudentToClass(studentName = String) {
+
+let message = "Only strings are allowed."
+    if(typeof studentName !=='string'){
+        return message;
+    }else if(studentName !== " " && !class07Students.find(name => name === studentName && getNumberOfStudents.length < 6)){
```

Comment:
Good job on this implementation! I especially like your validation that a string was actually provided! ⭐ 

`.find` is a somewhat advanced feature, can you explain what it does in this case and what `=>` means here? 😄 Can you find another method for arrays that could simplify this? (Hint: jump over to [MDN's page on arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array))

Something weird happens when you do `addStudentToClass("")` and nothing stops me from keeping to add people after the limit has been reached. Can you see why?  🐛 

There are also some duplicate code, i.e., line 7 shares some very similar code to line 13. Can you reorder your if-else-if-else such that this is not needed? This will likely also help with the bugs mentioned above! 😄

If you struggle with getting this to work, feel free to shoot me a message on Slack! 📬

Reply:
yes you are right
went through the code again. i found many mistakes
i added  1 more condition studentName === ""
i changed the whole code. 
thanks for the your review.